plugins/qiskit_executor/tasks.py

```diff
@@ -191,21 +114,113 @@ def execution_task(self, db_id: int) -> str:
             )
             execution_options.update(options)
 
-    if ibmq_token == "****":
+    if circuit_params.ibmqToken == "****":
         TASK_LOGGER.info("Loading IBMQ token from environment variable")
 
         if "IBMQ_TOKEN" in os.environ:
-            ibmq_token = os.environ["IBMQ_TOKEN"]
+            circuit_params.ibmqToken = os.environ["IBMQ_TOKEN"]
             TASK_LOGGER.info("IBMQ token successfully loaded from environment variable")
         else:
             TASK_LOGGER.info("IBMQ_TOKEN environment variable not set")
 
-    backend = backend_qiskit.get_qiskit_backend(ibmq_token, custom_backend)
-    backend.shots = shots
+    backend = get_qiskit_backend(circuit_params.backend, circuit_params.ibmqToken)
+    if backend is None:
+        msg = f"Could not load backend {circuit_params.backend}!"
+        TASK_LOGGER.error(msg)
+        raise ValueError(msg)
+    backend.shots = circuit_params.shots
+
+    circuit = QuantumCircuit.from_qasm_str(circuit_qasm)
+
+    job: IBMQJob = execute(circuit, backend, shots=execution_options["shots"])
 
-    metadata, counts, state_vector = execute_circuit(
-        circuit_qasm, backend, execution_options
+    db_task.data = dumps(
```

Comment:
`db_task.data` should directly support storing json compatible objects like dicts without converting them to strings.